### PR TITLE
[Behat] Add state transitions to creating orders programmatically

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Setup/OrderContext.php [hunk 1]
```diff
@@ -239,14 +245,15 @@ public function theCustomerChoseShippingWithPayment(
 
         $this->orderShipmentFactory->processOrderShipment($order);
         $order->getShipments()->first()->setMethod($shippingMethod);
-
-        $this->orderRecalculator->recalculate($order);
+
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/OrderContext.php]: redundant

File: src/Sylius/Behat/Context/Setup/OrderContext.php [hunk 2]
```diff
@@ -240,13 +238,15 @@ public function theCustomerChoseShippingWithPayment(
...
-        $this->orderRecalculator->recalculate($order);
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
 
         $payment = $this->paymentFactory->createWithAmountAndCurrencyCode($order->getTotal(), $order->getCurrencyCode());
         $payment->setMethod($paymentMethod);
-
         $order->addPayment($payment);
 
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_SELECT_PAYMENT);
+        $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply(OrderCheckoutTransitions::TRANSITION_COMPLETE);
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Setup/OrderContext.php]: Maybe we could extract this all transitions calls to private method? Would be much cleaner:
```php
private function applyTransition(OrderInterface $order, $transition)
{
    $this->stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->apply($transition);
}
```
wdyt?

Comment (GSadee) [inline src/Sylius/Behat/Context/Setup/OrderContext.php]: Not a bad idea :)

**Comment by TheMadeleine** [comment]
*[Behat] Add state transitions to creating orders programmatically

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

**Pull request merged**